Helper/Inventory.php

```diff
@@ -126,11 +127,50 @@ private function getQuantityAndStockStatusWithMSI(ProductModel $product, ?int $s
     {
         $stockItemData = $this->getStockItemData($product->getSku(), $stockId);
         $qty = $stockItemData[GetStockItemDataInterface::QUANTITY];
-        $availability = $stockItemData[GetStockItemDataInterface::IS_SALABLE];
+        $availability = $this->isProductSalable($product, $stockId);
 
         return [$qty, $availability];
     }
 
+    /**
+     * Get info about the salability of a product
+     * If a product is a grouped product, we consider it is salable
+     * if any of its associated products is salable
+     *
+     * @param ProductModel $product
+     * @param int|null $stockId
+     *
+     * @return boolean
+     */
+    private function isProductSalable(ProductModel $product, ?int $stockId = null)
+    {   
+        if ($product->getTypeId() == Grouped::TYPE_CODE) {
+            $associatedProducts = $product->getTypeInstance()->getAssociatedProducts($product);
+            foreach ($associatedProducts as $associatedProduct) {
+                if ($this->isNonGroupedProductSalable($associatedProduct, $stockId)) {
+                    return true;
+                }
+            }
+            return false;
+        }
+        
+        return $this->isNonGroupedProductSalable($product, $stockId);
+    }
+
+    /**
+     * Get info about the salability of any product that is not grouped
+     *
+     * @param ProductModel $product
+     * @param int|null $stockId
+     *
+     * @return boolean
+     */
+    private function isNonGroupedProductSalable(ProductModel $product, ?int $stockId = null)
```

Comment:
creo que el nombre se podría simplificar porque al final lo que se quiere mirar es si un producto es vendible no?
```suggestion
    private function isProductSalable(ProductModel $product, ?int $stockId = null)
```

Reply:
ese nombre ya lo tiene la función de arriba, no se podría poner no?

Reply:
sí, pon otro :+1:  pero busca un nombre que se entienda que es para obtener si es salable o no de un producto unico